scaleway/resource_mnq_credential.go

```diff
@@ -0,0 +1,268 @@
+package scaleway
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	mnq "github.com/scaleway/scaleway-sdk-go/api/mnq/v1alpha1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+const (
+	defaultMNQCredentialTimeout = 5 * time.Minute
+)
+
+func resourceScalewayMNQCredential() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceScalewayMNQCredentialCreate,
+		ReadContext:   resourceScalewayMNQCredentialRead,
+		UpdateContext: resourceScalewayMNQCredentialUpdate,
+		DeleteContext: resourceScalewayMNQCredentialDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create:  schema.DefaultTimeout(defaultMNQCredentialTimeout),
+			Read:    schema.DefaultTimeout(defaultMNQCredentialTimeout),
+			Update:  schema.DefaultTimeout(defaultMNQCredentialTimeout),
+			Delete:  schema.DefaultTimeout(defaultMNQCredentialTimeout),
+			Default: schema.DefaultTimeout(defaultMNQCredentialTimeout),
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Optional:    true,
+				Description: "The name of the Credential",
+			},
+			"namespace_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
```

Comment:
[`namespace_id` is required](https://developers.scaleway.com/en/products/messaging_and_queuing/api/v1alpha1/#post-67608e)